packages/two_dimensional_scrollables/lib/src/tree_view/render_tree.dart

```diff
@@ -0,0 +1,744 @@
+/// displays a subset of its [TreeViewNode] rows according to its own dimensions
+/// and the given [verticalOffset] and [horizontalOffset]. As the offset varies,
+/// different nodes are visible through the viewport.
+class RenderTreeViewport extends RenderTwoDimensionalViewport {
+  /// Creates a viewport for [RenderBox] objects in a tree format of rows.
+  RenderTreeViewport({
+    required Map<UniqueKey, TreeViewNodesAnimation> activeAnimations,
+    required Map<int, int> rowDepths,
+    required TreeViewTraversalOrder traversalOrder,
+    required double indentation,
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required TreeRowDelegateMixin super.delegate,
+    required super.childManager,
+    super.cacheExtent,
+    super.clipBehavior,
+  })  : _activeAnimations = activeAnimations,
+        _rowDepths = rowDepths,
+        _traversalOrder = traversalOrder,
+        _indentation = indentation,
+        assert(verticalAxisDirection == AxisDirection.down &&
+            horizontalAxisDirection == AxisDirection.right),
+        super(
+          mainAxis: traversalOrder == TreeViewTraversalOrder.depthFirst
+              ? Axis.vertical
+              : Axis.horizontal,
+        );
+
+  @override
+  TreeRowDelegateMixin get delegate => super.delegate as TreeRowDelegateMixin;
+  @override
+  set delegate(TreeRowDelegateMixin value) {
+    super.delegate = value;
+  }
+
+  /// The currently active [TreeViewNode] animations.
+  ///
+  /// Since the index of animating nodes can change at any time, the unique key
+  /// is used to track an animation of nodes across frames.
+  Map<UniqueKey, TreeViewNodesAnimation> get activeAnimations {
+    return _activeAnimations;
+  }
+
+  Map<UniqueKey, TreeViewNodesAnimation> _activeAnimations;
+  set activeAnimations(Map<UniqueKey, TreeViewNodesAnimation> value) {
+    if (_activeAnimations == value) {
+      return;
+    }
+    _activeAnimations = value;
+    markNeedsLayout(withDelegateRebuild: true);
+  }
+
+  /// The depth of each currently active node in the tree.
+  ///
+  /// This is used to properly set the [TreeVicinity] for the [traversalOrder].
+  Map<int, int> get rowDepths => _rowDepths;
+  Map<int, int> _rowDepths;
+  set rowDepths(Map<int, int> value) {
+    if (_rowDepths == value) {
+      return;
+    }
+    _rowDepths = value;
+    markNeedsLayout();
+  }
+
+  /// The order in which child nodes of the tree will be traversed.
+  ///
+  /// The default traversal order is [TreeViewTraversalOrder.depthFirst].
+  TreeViewTraversalOrder get traversalOrder => _traversalOrder;
+  TreeViewTraversalOrder _traversalOrder;
+  set traversalOrder(TreeViewTraversalOrder value) {
+    if (_traversalOrder == value) {
+      return;
+    }
+    _traversalOrder = value;
+    // We don't need to layout again. This is used when we visit children.
```

Comment:
It says that this is used when visiting children, but this class doesn't seem to use `traversalOrder` for anything (expect initially in the constructor to determine the mainAxis). Is there some implementation missing?

Reply:
The traversal is handled in the TwoDimensionalViewport, as dictated by mainAxis. :)